src/index.js

```diff
@@ -544,7 +544,7 @@ function Component(props, context, opts) {
 		newComponentHook.call(this, props, context);
 	}
 }
-extend(Component.prototype = new PreactComponent(), {
+extend(Component.prototype = PreactComponent.prototype, {
```

Comment:
this will mutate the prototype of preact's `Component`.

Instead we'll have to use Object.create or something.